Detectors/EMCAL/simulation/src/Detector.cxx

```diff
@@ -135,101 +155,145 @@ void Detector::ConstructGeometry()
-  // EMCAL geometry implementation in VMC works with a copy of the volume and placing it n-times into the mother volume
-  // via translation / rotation, so the copy index is the index of a tower / module / supermodule node within a mother
-  // volume Additional care needs to be taken for the supermodule index: The copy is connected to a certain supermodule
-  // type, which differs for various parts of the detector
-  Int_t copyEta, copyPhi, copyMod, copySmod;
-  fMC->CurrentVolID(copyEta);       // Tower in module - x-direction
-  fMC->CurrentVolOffID(1, copyPhi); // Tower in module - y-direction
-  fMC->CurrentVolOffID(3, copyMod); // Module in supermodule
-  fMC->CurrentVolOffID(
-    4, copySmod); // Supermodule in EMCAL - attention, with respect to a given supermodule type (offsets needed)
-  std::string smtype(fMC->CurrentVolOffName(4));
-  int offset(0);
-  if (smtype == "SM3rd") {
-    offset = 10;
-  } else if (smtype == "DCSM") {
-    offset = 12;
-  } else if (smtype == "DCEXT") {
-    offset = 18;
-  }
-  LOG(DEBUG3) << "Supermodule copy " << copySmod << ", module copy " << copyMod << ", y-dir " << copyPhi << ", x-dir "
-              << copyEta << ", supermodule ID " << copySmod + offset - 1;
-  LOG(DEBUG3) << "path " << fMC->CurrentVolPath();
-  LOG(DEBUG3) << "Name of the supermodule type " << fMC->CurrentVolOffName(4) << ", Module name "
-              << fMC->CurrentVolOffName(3);
-
-  // porting from aliroot: circumvention of the index problem. GetAbsCellID cannot be used
-  // Int_t detID = geom->GetAbsCellId(offset + copySmod - 1, copyMod - 1, copyPhi - 1, copyEta - 1);
-  //
-  // Swap A side in Phi and C side in Eta due to wrong indexing.
-  //
-  Int_t smNumber = offset + copySmod - 1, smTypeID = 1;
-  auto [iphi, ieta] = geom->GetCellPhiEtaIndexInSModule(smNumber, copyMod - 1, copyPhi - 1, copyEta - 1);
-  //iphi = std::get<0>(posetaphi);
-  //ieta = std::get<1>(posetaphi);
-  if (smNumber % 2 == 0) {
-    if (strcmp(fMC->CurrentVolOffName(4), "DCSM") == 0)
-      smTypeID = 3; //DCal supermodule. previous design/idea
-    else
-      smTypeID = 2;
-    ieta = ((geom->GetCentersOfCellsEtaDir()).size() * 2 / smTypeID - 1) - ieta; // 47/31-ieta, revert the ordering on A side in order to keep convention.
-  } else {
-    if (strcmp(fMC->CurrentVolOffName(4), "SM10") == 0)
-      smTypeID = 2; //half supermodule. previous design/idea
-    if (strcmp(fMC->CurrentVolOffName(4), "SM3rd") == 0)
-      smTypeID = 3; //one third (installed in 2012) supermodule
-    if (strcmp(fMC->CurrentVolOffName(4), "DCEXT") == 0)
-      smTypeID = 3;                                                          //one third (installed in 2012) supermodule
-    iphi = ((geom->GetCentersOfCellsPhiDir()).size() / smTypeID - 1) - iphi; // 23/7-iphi, revert the ordering on C side in order to keep convention.
+  auto track = fMC->GetStack()->GetCurrentTrackNumber(),
+       directparent = fMC->GetStack()->GetCurrentParentTrackNumber();
```

Comment:
would prefer another "auto" instead of ",". It looks weird.

Reply:
I can replace the auto by the direct type int, however I would prefer then to have the "," instead of a new type declaration.